common/transport/mqtt/src/mqtt_base.ts

```diff
@@ -229,6 +229,10 @@ export class MqttBase extends EventEmitter {
           agent: this._options.mqtt.webSocketAgent
         };
       }
+      if (this._options.productInfo) {
+        // According to CUA Spec: "Append entire string to Username field in CONNECT packet" 
+        options.username += this._options.productInfo
```

Comment:
Do we need to do anything to url encode this?

Reply:
really good question - given the username isn't used as a URL i would assume no... but it might be worth double-checking with someone in the service team

---
In reply to: [295432142](https://github.com/Azure/azure-iot-sdk-node/pull/572#discussion_r295432142) [](ancestors = 295432142)

Reply:
So... I've updated this. Background: I thought that I should appending the productInfo to the options.username.... Essentially tagging it to the mqtt username. BUT then I dug deeper and I think for the productInfo rather than adding it to the username I should just append it to however we already do the userAgent.